packages/backend-function/src/factory.ts

```diff
@@ -31,17 +31,25 @@ import { FunctionEnvironmentTypeGenerator } from './function_env_type_generator.
 import { AttributionMetadataStorage } from '@aws-amplify/backend-output-storage';
 import { fileURLToPath } from 'node:url';
 import { AmplifyUserError, TagName } from '@aws-amplify/platform-core';
+import * as lambda from 'aws-cdk-lib/aws-lambda';
```

Comment:
nit: prefer `import { Function } from 'aws-cdk-lib/aws-lambda'`

also, we should probably be using the `IFunction` interface here rather than the `Function` base class